enterprise/javaee.wildfly/src/org/netbeans/modules/javaee/wildfly/ide/WildflyJ2eePlatformFactory.java

```diff
@@ -334,21 +334,13 @@ String getDefaultJpaProvider() {
         }
 
         private boolean containsJaxWsLibraries() {
-            File[] jaxWsAPILib = new File(properties.getModulePath("org/jboss/ws/api/main")).listFiles(new FilenameFilter() {
-                @Override
-                public boolean accept(File dir, String name) {
-                    return name.startsWith("jbossws-api") && name.endsWith("jar");
-                }
-            }); // NOI18N
+            File[] jaxWsAPILib = new File(properties.getModulePath("org/jboss/ws/api/main"))                       // NOI18N
+                    .listFiles((File dir, String name) -> name.startsWith("jbossws-api") && name.endsWith("jar")); // NOI18N
             if (jaxWsAPILib != null && jaxWsAPILib.length == 1 && jaxWsAPILib[0].exists()) {
```

Comment:
Is there a possibility that `listFiles` returns a non existent file? So is the check for existence necessary?

Reply:
My reading is, that the null check is required, but the existence check is not necessary. However I don't expect this code to be a hot path, so it should be harmless.